internal/util/util.go

```diff
@@ -360,6 +360,64 @@ func GetVolumeSnapshotbackupWithStatusData(volumeSnapshotbackupNS string, volume
 	return vsb, nil
 }
 
+// Get VolumeSnapshotBackup CR with status data
+func GetVolumeSnapshotRestoreWithStatusData(restoreName string, PVCName string, log logrus.FieldLogger) (datamoverv1alpha1.VolumeSnapshotRestoreList, error) {
+
+	vsrList := datamoverv1alpha1.VolumeSnapshotRestoreList{}
+	// default timeout value is 10
+	timeoutValue := "10m"
+	// use timeout value if configured
+	if len(os.Getenv(DatamoverTimeout)) > 0 {
+		timeoutValue = os.Getenv(DatamoverTimeout)
+	}
+
+	timeout, err := time.ParseDuration(timeoutValue)
+	if err != nil {
+		return vsrList, errors.Wrapf(err, "error parsing the datamover timout")
+	}
+	interval := 5 * time.Second
+
+	////
+	err = wait.PollImmediate(interval, timeout, func() (bool, error) {
+
+		snapMoverClient, err := GetVolumeSnapshotMoverClient()
+		if err != nil {
+			return false, err
+		}
+
+		VSRListOptions := client.MatchingLabels(map[string]string{
+			velerov1api.RestoreNameLabel: restoreName,
+			PersistentVolumeClaimLabel:   PVCName,
+		})
+
+		err = snapMoverClient.List(context.TODO(), &vsrList, VSRListOptions)
+		if err != nil {
+			return false, errors.Wrapf(err, fmt.Sprintf("failed to get volumesnapshotrestoreList for PVC %s", PVCName))
+		}
+
+		if len(vsrList.Items) > 0 {
+			if len(vsrList.Items[0].Status.SnapshotHandle) == 0 || len(vsrList.Items[0].Status.Phase) == 0 {
```

Comment:
Testing this and discovered that if validation fails, VSR only has `vsr.status.phase` data, so the above keeps polling, even if VSR is failed. Anyway these conditions could be switched?  May also need a nil check for `vsr.status`

Reply:
switched the if conditionals, re: nil check for status, we can't do that as its not a pointer, hence the status data conditional